### PR TITLE
[Doc] Add consumer configuration description for Java client

### PR DESCRIPTION
**Pull request by Anonymitaet**

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@yjshen could you please help review? Many thanks

I don't find the following parameter descriptions in the Pulsar project, could you please help add? Thank you

* acknowledgementsGroupTimeMicros
* negativeAckRedeliveryDelayMicros
* tickDurationMillis
* properties
* readCompacted
* patternAutoDiscoveryPeriod
* autoUpdatePartitions
* replicateSubscriptionState
* resetIncludeHead

**Comment by Anonymitaet** [comment]
Resolve https://github.com/apache/pulsar/issues/4509

**Comment by Anonymitaet** [comment]
@yjshen thanks for your technical guidance.

I've incorporated your comments, could you PTAL?

File: site2/docs/client-libraries-java.md [hunk 1]
```diff
@@ -247,7 +247,36 @@ while (true) {
...
+boolean|`autoUpdatePartitions`|If `autoUpdatePartitions` is enabled, a consumer subscribes to partition increasement automatically.<br/><br/>**Note**: this is only for partitioned consumers.|true
+boolean|`replicateSubscriptionState`|If `replicateSubscriptionState` is enabled, a subscription state is replicated to geo-replicated clusters.|false
+boolean|`resetIncludeHead`|None|false
```
Comment (yjshen) [inline site2/docs/client-libraries-java.md]: Pls remove this line, it's not a consumer-related configuration.

Comment (Anonymitaet) [reply site2/docs/client-libraries-java.md]: OK

**Comment by yjshen** [comment]
run java8 tests

**Comment by yjshen** [comment]
run java8 tests

**Comment by yjshen** [comment]
run java8 tests

**Pull request merged**